### PR TITLE
[Refactor:System] Update Node Repo

### PR DESCRIPTION
**Pull request by cjreed121**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
There is currently a deprecation in the node installation script. This currently halts installation for a minute and this method will likely be removed in the future.

### What is the new behavior?
Rather than fetching the nodesource script, the debian repo is inserted manually. This is the new suggested way to install node. This PR also bumps Node 16 to 20.

### Other information?
Testing vagrant up here: https://github.com/Submitty/Submitty/actions/runs/6149685803

### CODE REVIEW

File: migration/migrator/migrations/system/20230905224747_update_node_repo.py [hunk 1]
```diff
@@ -26,4 +26,9 @@ def down(config):
     :param config: Object holding configuration details about Submitty
     :type config: migrator.config.Config
     """
-    pass
+    os.system("apt-get purge nodejs libnode72 -y")
+    os.system("rm -r /etc/apt/sources.list.d/nodesource.list")
+    os.system("rm -r /etc/apt/keyrings/nodesource.gpg")
+    os.system("umask 0022 && curl -sL https://deb.nodesource.com/setup_16.x | bash -")
```
Comment (williamjallen) [inline migration/migrator/migrations/system/20230905224747_update_node_repo.py:32]: Should we also bump nodejs while we're at it?

Comment (cjreed121) [reply migration/migrator/migrations/system/20230905224747_update_node_repo.py:32]: Yeah we probably should

**Review by williamjallen — APPROVED** [review]
Changes seem reasonable to me.  It's worth noting that last time we bumped the nodejs version, we used a [sysadmin action](https://submitty.org/sysadmin/installation/version_notes/v21.11.01) rather than a migration.  If Barb is fine with using a migration for this, this should be ready to merge.

**Review by ziesski — APPROVED** [review]
I tested it out, and it fixed my vagrant up. Thanks :)

**Pull request merged**